From 021af6dcf439d70df46b9fc839c2cc0877b8e43c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Thu, 28 Oct 2021 09:11:02 +0200
Subject: [PATCH] server: Clear the MOUSEEVENTF_(ABSOLUTE|VIRTUALDESK) flags.

For rawinput messages, as user32 is currently only expecting relative
motion.
---
 server/queue.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/queue.c b/server/queue.c
index 1aedda1ca71..87b25820cfe 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -1967,7 +1967,7 @@ static int queue_mouse_message( struct desktop *desktop, user_handle_t win, cons
         raw_msg.message    = WM_INPUT;
 
         raw_msg.info                = input->mouse.info;
-        raw_msg.flags               = flags;
+        raw_msg.flags               = flags & ~(MOUSEEVENTF_ABSOLUTE | MOUSEEVENTF_VIRTUALDESK);
         raw_msg.rawinput.type       = RIM_TYPEMOUSE;
         raw_msg.rawinput.mouse.x    = x - desktop->cursor.x;
         raw_msg.rawinput.mouse.y    = y - desktop->cursor.y;
-- 
2.43.0

